ocaml/xapi-idl/cluster/cluster_interface.ml

```diff
@@ -271,4 +272,129 @@ module LocalAPI (R : RPC) = struct
       @-> enabled_p
       @-> returning unit_p err
       )
+
+  type dict = (string * string) list [@@deriving rpcty]
+
+  module TypeCombinators = struct
+    let option ?name ?(description = []) d =
+      let open Rpc.Types in
+      let name =
+        match name with
+        | Some n ->
+            n
+        | None ->
+            Printf.sprintf "%s option" d.name
+      in
+      {name; description; ty= Option d.ty}
+
+    let list ?name ?(description = []) d =
+      let open Rpc.Types in
+      let name =
+        match name with
+        | Some n ->
+            n
+        | None ->
+            Printf.sprintf "list of %ss" d.name
+      in
+      {name; description; ty= List d.ty}
+
+    let pair ?name ?(description = []) (p0, p2) =
+      let open Rpc.Types in
+      let name =
+        match name with
+        | Some n ->
+            n
+        | None ->
+            Printf.sprintf "pair of %s and %s" p0.name p2.name
+      in
+      {name; description; ty= Tuple (p0.ty, p2.ty)}
+
+    let triple ?name ?(description = []) (p1, p2, p3) =
+      let open Rpc.Types in
+      let name =
+        match name with
+        | Some n ->
+            n
+        | None ->
+            Printf.sprintf "triple of %s, %s and %s" p1.name p2.name p3.name
+      in
+      {name; description; ty= Tuple3 (p1.ty, p2.ty, p3.ty)}
+  end
+
+  module Observer = struct
+    open TypeCombinators
+
+    let endpoints_p = Param.mk ~name:"endpoints" (list Types.string)
+
+    let bool_p = Param.mk ~name:"bool" Types.bool
```

Comment:
Parameters that are named after their type are not very descriptive. It is better to give them more meaningful names. For these one, however, I realise that you have copied the code from xenops_interface.ml.